api.go

```diff
@@ -70,33 +70,40 @@ type ClientPersistent struct {
 	client        *Client
 }
 
-// GetVersion returns version
-func (c *Client) GetVersion() (string, error) {
-	resp, err := c.api.DoAndGetResponseBody(
+func (c *Client) GetVersion() (ver string, err error) {
+	var resp *http.Response
+	fmt.Println(c.GetToken())
+	resp, err = c.api.DoAndGetResponseBody(
 		context.Background(), http.MethodGet, "/api/version", nil, nil, c.configConnect.Version)
 	if err != nil {
 		return "", err
 	}
-
 	defer func() {
 		if err := resp.Body.Close(); err != nil {
 			doLog(log.WithError(err).Error, "")
 		}
 	}()
-
 	// parse the response
 	switch {
 	case resp == nil:
 		return "", errNilReponse
-	case !(resp.StatusCode >= 200 && resp.StatusCode <= 299):
+	case resp.StatusCode == 401:
+		// Authenticate then try again
+		if _, err = c.Authenticate(c.configConnect); err != nil {
+			return "", c.api.ParseJSONError(resp)
```

Comment:
@KshitijaKakde How about directly returning the error here like line no. 78-79 after logging it.
```
if err != nil {
		return "", err
	}
```

In case if you want to parse the resp body, add a check and make sure that resp !=nil
```
resp, err := c.api.DoAndGetResponseBody(
    context.Background(), http.MethodGet, "/api/version", nil, nil, c.configConnect.Version)
if err != nil {
    if resp != nil {
        return "", c.api.ParseJSONError(resp)
    }
    return "", err
}

```